packages/actions/src/internal/decodeRevertReason.js

```diff
@@ -0,0 +1,74 @@
+import { decodeAbiParameters, hexToString } from 'viem'
+
+/**
+ * Decodes basic revert data similar to Anvil's approach
+ * @param {import('@tevm/utils').Hex | string | undefined} data - The revert data
+ * @returns {string} - The decoded reason or 'execution reverted' if not decodable
+ */
+export const decodeRevertReason = (data) => {
```

Comment:
There is a lot of logic here to do this does this functionality already exist in viem or ox? If so we should use their implementation rather than implementing all this ourselves.

If it's not in ox we should open an issue in ox to add these utilities and add a TODO here to replace it with ox